sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusReceivedMessage.cs

```diff
@@ -36,6 +39,7 @@ internal ServiceBusReceivedMessage(ReadOnlyMemory<byte> body)
         internal ServiceBusReceivedMessage(AmqpAnnotatedMessage message)
         {
             AmqpMessage = message;
+            body = new Lazy<BinaryData>(() => AmqpMessage.GetBody(), LazyThreadSafetyMode.ExecutionAndPublication);
```

Comment:
Used the `LazyThreadSafetyMode.ExecutionAndPublication` to make sure when it throws the behavior is as before